docs(manager/dockerfile): Document supported dependencies for Dockerfile manager (#32288)
renovatebot/renovate: lib/modules/manager/dockerfile/readme.md

### Supported dependencies

This manager extracts image references in a `Dockerfile` and/or `Containerfile` and supports:

> 1. [`FROM`](https://docs.docker.com/reference/dockerfile/#from) images

Reviewer: ```suggestion
- [`FROM`](https://docs.docker.com/reference/dockerfile/#from) images
```

> 1. [`COPY --from`](https://docs.docker.com/reference/dockerfile/#copy---from) images

Reviewer: ```suggestion
- [`COPY --from`](https://docs.docker.com/reference/dockerfile/#copy---from) images
```

> 1. [`syntax`](https://docs.docker.com/reference/dockerfile/#syntax) images

Reviewer: ```suggestion
- [`syntax`](https://docs.docker.com/reference/dockerfile/#syntax) images
```

#### `FROM` support

> Renovate can update images referenced in `FROM` directives. This also works for multi stage builds where multiple `FROM` directives are being used in a single `Dockerfile`.

Reviewer: ```suggestion
Renovate can update images referenced in `FROM` directives.
This even works for multi-stage builds with many `FROM` directives in one Dockerfile.
```

```dockerfile
FROM node:20.9.0
```

Advanced `FROM` flags like `--platform` or `AS <name>` are also supported:

```dockerfile
FROM --platform=linux/amd64 node:20.9.0 AS installer
```

> Also variables and [`ARG` directives](https://docs.docker.com/reference/dockerfile/#understand-how-arg-and-from-interact) will automatically be expanded:

Reviewer: ```suggestion
Also, Renovate will automatically expand variables and [`ARG` directives](https://docs.docker.com/reference/dockerfile/#understand-how-arg-and-from-interact):
```
